pd/src/consensus/worker.rs

```diff
@@ -247,164 +241,235 @@ impl Worker {
             .height
             .try_into()
             .expect("height should be nonnegative");
-        let epoch = pending_block.set_height(height);
+        let epoch = pending_block.set_height(
+            height,
+            self.state
+                .private_reader()
+                .chain_params_rx()
+                .borrow()
+                .epoch_duration,
+        );
 
         tracing::debug!(?height, ?epoch, end_height = ?epoch.end_height());
 
+        // Find out which validators were slashed in this block
+        let slashed_validators = pending_block
+            .validator_state_changes
+            .iter()
+            .filter(|p| matches!(p.1, ValidatorState::Slashed))
```

Comment:
We filter out validators who were slashed in *this block* here: this is correct because we don't need to worry about previous blocks' slashings, since we should already have removed their quarantined items from the database during the commit of that block.

Reply:
I don't follow -- isn't this filtering *in* validators who were slashed in this block, not filtering them out?

Reply:
(The code looks correct, just confused by your review comment)